app/__init__.py

```diff
@@ -221,7 +225,158 @@ def linchpin_fetch_workspace() -> Response:
+
+
+@app.route('/api/v1.0/workspaces/up', methods=['POST'])
+def linchpin_up() -> Response:
+    """
+        POST request route for provisioning workspaces/pinFile already created
+        RequestBody: {"id": "workspace_id",
+                    provision_type: "workspace",
+                    --> value can be either pinfile or workspace
+                    }
+        :return : response with provisioned workspace id, status,
+                  contents_of_latest_inventory_generated_in_inventoryfolder,
+                  contents_of_linchpin.latest_file_in_resource_folder
+    """
+    identity = None
+    try:
+        data = request.json  # Get request body
+        provision_type = data['provision_type']
+        if provision_type == "workspace":
+            identity = data['id']
+            if not os.path.exists(WORKING_PATH + "/" + identity):
+                return jsonify(status=response.NOT_FOUND)
+            cmd = create_cmd_workspace(data, identity, "up")
+        elif provision_type == "pinfile":
+            if 'name' in data:
+                identity = str(uuid.uuid4()) + "_" + data['name']
+            else:
+                identity = str(uuid.uuid4())
+            precmd = ["linchpin", "-w " + WORKING_DIR + identity +
+                      "/", "init"]
+            output = subprocess.Popen(precmd, stdout=subprocess.PIPE)
+            output.communicate()
+            get_connection().db_insert_no_name(identity,
+                                               response.WORKSPACE_REQUESTED)
+            cmd = create_cmd_up_pinfile(data, identity)
+        else:
+            raise ValueError
+        output = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        output.communicate()
+        linchpin_latest_path = WORKING_PATH + "/" + identity + LATEST_PATH
+        with open(linchpin_latest_path, 'r') as file:
+            linchpin_latest = json.load(file)
+        directory_path = glob.glob(WORKING_PATH + "/" + identity +
+                                   INVENTORY_PATH)
+        latest_file = max(directory_path, key=os.path.getctime)
+        with open(latest_file, 'r') as data:
+            inventory = data.read().replace('\n', ' ')
+        get_connection().db_update(identity, response.PROVISION_STATUS_SUCCESS)
+        return jsonify(id=identity,
+                       status=response.PROVISION_SUCCESS,
+                       inventory=inventory,
+                       latest=linchpin_latest,
+                       code=output.returncode,
+                       mimetype='application/json')
+    except (KeyError, ValueError, TypeError):
+        return jsonify(status=errors.ERROR_STATUS,
+                       message=errors.KEY_ERROR_UP)
+    except Exception as e:
+        get_connection().db_update(identity, response.PROVISION_FAILED)
+        app.logger.error(e)
+        return jsonify(status=errors.ERROR_STATUS, message=str(e))
+
+
+@app.route('/api/v1.0/workspaces/destroy', methods=['POST'])
```

Comment:
would not these be just up and destroy instead of /workspaces/up and /workspaces.destroy
since up is not specific to workspaces we can use pin files too
@mansikulkarni96

Reply:
We can provide a pinfile, but we create a workspace from it eventually and operate on the workspace while doing linchpin up/destroy, hence I named it in this way. Although it's true that we can provide pinfile and workpsace both, so it makes sense to name /up as well.